### ISSUE TITLE
[doc] Moving away from LaTeX

### ISSUE THREAD

**Issue by ejgallego** [coq/coq#7390]
After the completion of the Sphinx migration [#6818], there are only residual TeX files in the tree; we all tend to agree that they would rather be moved to other formats. In today's `master` we can see the remaining files:

```
egallego@rochefort$ find . -name *.tex
./vernac/doc.tex
./interp/doc.tex
./lib/doc.tex
./pretyping/doc.tex
./proofs/doc.tex
./library/doc.tex
./tactics/doc.tex
./kernel/doc.tex
```
I will take care of removing these once #7155 lands, I will move the corresponding text there to the proper files. We are waiting on a new `odoc` upstream release for that tho + Dune.
```
./dev/v8-syntax/syntax-v8.tex
./dev/v8-syntax/memo-v8.tex
./dev/ocamldoc/header.tex
./dev/doc/naming-conventions.tex
./dev/doc/versions-history.tex
./dev/doc/newsyntax.tex
./dev/doc/minicoq.tex
```
No idea on what to do with these files? Should be converted to MarkDown / RST ?
```
./doc/tools/Translator.tex
./doc/common/title.tex
./doc/common/version.tex
./doc/common/macros.tex
./doc/stdlib/Library.coqdoc.tex
./doc/stdlib/Library.tex
./doc/tutorial/Tutorial.tex
./doc/tutorial/Tutorial.v.tex
./doc/whodidwhat/whodidwhat-8.5update.tex
./doc/whodidwhat/whodidwhat-8.2update.tex
./doc/whodidwhat/whodidwhat-8.4update.tex
./doc/whodidwhat/whodidwhat-8.3update.tex
./doc/RecTutorial/recmacros.tex
./doc/RecTutorial/RecTutorial.tex
./doc/RecTutorial/coqartmacros.tex
```
I guess a few of these files should be ported to Sphinx; dunno for the rest.

**Comment by SkySkimmer** [comment]
I don't know how coqdoc works, does it go through latex when producing other formats?

**Comment by ejgallego** [comment]
> I don't know how coqdoc works, does it go through latex when producing other formats?

I does not.

**Comment by ejgallego** [comment]
Note that the test suite does indeed depend on LaTeX so we are not gonna be moving so fast from it [unless we only test the html targets].

**Comment by Zimmi48** [comment]
I would already be interested in having less dependencies for users who do not run the test-suite.

**Comment by Zimmi48** [comment]
To answer your initial questions:

```
./dev/v8-syntax/syntax-v8.tex
./dev/v8-syntax/memo-v8.tex
./dev/ocamldoc/header.tex
./dev/doc/naming-conventions.tex
./dev/doc/versions-history.tex
./dev/doc/newsyntax.tex
./dev/doc/minicoq.tex
```

These should either be converted to Markdown or kept as `.tex` but moved to a `history/` folder (see also #7033). AFAIK these files are not included in any Makefile anyways so we can consider removing the dependency on LaTeX even without converting those.

```
./doc/tools/Translator.tex
```

Probably outdated doc for tool that doesn't exist anymore. But @herbelin should know more.

```
./doc/common/title.tex
./doc/common/version.tex
./doc/common/macros.tex
```

Probably unused remaining bits of the LaTeX reference manual. @maximedenes?

```
./doc/stdlib/Library.coqdoc.tex
./doc/stdlib/Library.tex
```

Is there a target to build a PDF version of the stdlib doc? If yes, that's probably part of it. If not, that's probably dead code...

```
./doc/tutorial/Tutorial.tex
./doc/tutorial/Tutorial.v.tex
```

I was the last one to update this tutorial (for 8.7) and I think it is time to officially kill it. But at the minimum, it should be kept outside of the main repo. If it has to be maintained, it should probably be community-maintained.

```
./doc/whodidwhat/whodidwhat-8.5update.tex
./doc/whodidwhat/whodidwhat-8.2update.tex
./doc/whodidwhat/whodidwhat-8.4update.tex
./doc/whodidwhat/whodidwhat-8.3update.tex
```

As for the `tex` file in `dev/doc` they are probably not compiled anyways by any Makefile, so we can remove the LaTeX dependency without removing them. But removing these files completely should be done as part of a larger reflection on how to keep a memory of the work done by contributors (to consider together with the Credits part of the refman, the CREDITS file, the CHANGES file, the Zenodo records, the `README-V1-V5.asciidoc` file, the remaining special headers —see #7198—, etc.)

```
./doc/RecTutorial/recmacros.tex
./doc/RecTutorial/RecTutorial.tex
./doc/RecTutorial/coqartmacros.tex
```

To remove from the repo. If it has to be maintained, it should probably be community-maintained.

> Note that the test suite does indeed depend on LaTeX so we are not gonna be moving so fast from it [unless we only test the html targets].

There is a clarification to do about the test-suite dependencies as well. Currently, when some there are some missing dependencies, part of the test-suite are skipped silently, while others fail.

**Comment by ejgallego** [comment]
The plan sounds good.

**Comment by ejgallego** [comment]
@Zimmi48 should we close this as `wontfix` ?

**Comment by Zimmi48** [comment]
We could still move `dev/v8-syntax` and `doc/whodidwhat` to `dev/doc/archive` and we could stil remove the various `doc.tex` at the root of some components. But it's true that we won't go as far as removing our LaTeX dependency since we will continue to need it to build PDF versions of the doc.

**Comment by maximedenes** [comment]
> Probably unused remaining bits of the LaTeX reference manual. @maximedenes?

That doesn't ring a bell, sorry.

**Comment by herbelin** [comment]
> To answer your initial questions:
> 
> ```
> ./dev/v8-syntax/syntax-v8.tex
> ./dev/v8-syntax/memo-v8.tex
> ./dev/ocamldoc/header.tex
> ./dev/doc/naming-conventions.tex
> ./dev/doc/versions-history.tex
> ./dev/doc/newsyntax.tex
> ./dev/doc/minicoq.tex
> ```
> 
> These should either be converted to Markdown or kept as `.tex` but moved to a `history/` folder (see also #7033). AFAIK these files are not included in any Makefile anyways so we can consider removing the dependency on LaTeX even without converting those.

The file `versions-history.tex` is still relevant (and I consult it regularly!).

The file `naming-conventions.tex` failed in practice to trigger a dynamic on naming. It remains per se a good idea. That would be good that people having thought about naming (MathComp, Coquelicot, ...) share their experience in a collective place (if not already the case).

> ```
> ./doc/tools/Translator.tex
> ```
> 
> Probably outdated doc for tool that doesn't exist anymore. But @herbelin should know more.

Interesting for the history but not useful for the current archive. Idem for `newsyntax.tex` and the v8 files.

> ```
> ./doc/common/title.tex
> ./doc/common/version.tex
> ./doc/common/macros.tex
> ```
> 
> Probably unused remaining bits of the LaTeX reference manual. @maximedenes?

Were shared with the tutorial and PDF's library, probably.

OK with the other remarks.

**Comment by Zimmi48** [comment]
>Were shared with the tutorial and PDF's library, probably.

Indeed, and now they are probably only used by the PDF documentation of the stdlib.
